### ISSUE TITLE
Reproducing RoBERTa STS-B results

### ISSUE THREAD

**Issue by mitchellg** [pytorch/fairseq#1009]
Thanks for your great work! Quick request: could you please add how to replicate RoBERTa STS-B results? There are good instructions for fine-tuning the model but I didn't see any for running the prediction task itself. Thanks!

**Comment by ngoyal2707** [comment]
You can use something like following code-snippet for inference on `STS-B`.
```
from fairseq.models.roberta import RobertaModel
from scipy.stats import pearsonr

roberta = RobertaModel.from_pretrained(
    'checkpoints/',
    checkpoint_file='checkpoint_best.pt',
    data_name_or_path='STS-B-bin'
)

roberta.cuda()
roberta.eval()
gold, pred = [], []
with open('glue_data/STS-B/dev.tsv') as fin:
    fin.readline()
    for index, line in enumerate(fin):
        tokens = line.strip().split('\t')
        sent1, sent2, target = tokens[7], tokens[8], float(tokens[9])
        tokens = roberta.encode(sent1, sent2)
        features = roberta.extract_features(tokens)
        predictions = 5.0 * roberta.model.classification_heads['sentence_classification_head'](features)
        gold.append(target)
        pred.append(predictions.item())

print('| Pearson: ', pearsonr(gold, pred))
```

I will add above to readme, thanks for pointing it out.

**Comment by mitchellg** [comment]
Thanks so much! Unfortunately I wasn't able to reproduce your STS-B score. The value of predictions.item() was always ~2.84, changing only very slightly for each example. I followed your finetuning instructions for STS-B and didn't change the above code for inference. Any thoughts on what might be going on?

**Comment by ngoyal2707** [comment]
hmm can you please share your training logs? specially what loss you are getting?
Also can you please make sure you see "loaded checkpoint ..." file in the training logs?

**Comment by mitchellg** [comment]
You're right, I wasn't loading the checkpoint. Fixed that and now I'm able to reproduce STS-B. Thanks again!

**Comment by absin1** [comment]
Where do we get the checkpoint file?

**Comment by YJYJLee** [comment]
predictions = 5.0 * roberta.model.classification_heads['sentence_classification_head'](features)

Why do we need to multiply 5.0 to the output of the model?

**Comment by sandeep-adireddi** [comment]
why we need data_name_or_path as STS-bin

**Comment by QizhiPei** [comment]
> predictions = 5.0 * roberta.model.classification_heads['sentence_classification_head'](features)
> 
> Why do we need to multiply 5.0 to the output of the model?

Hello!
In the original STS problem, the lables range from 0 to 5. 
In the examples/roberta/preprocess_GLUE_tasks.sh(see [Line 182 183](https://github.com/pytorch/fairseq/blob/d792b793a777bf660d2aaeb095c2381af189e626/examples/roberta/preprocess_GLUE_tasks.sh?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L182)) the label is reranged to 0 to 1.
So in evaluation step, we need to  multiply 5.0 to the output of the model to compute the Pearson with original labels correctly.
